### PR TITLE
Labels.arange() function

### PR DESCRIPTION
**Pull request by frostedoyster**
----
:books: Documentation preview :books:: https://equistore--239.org.readthedocs.build/en/239/

### CODE REVIEW

File: python/src/equistore/labels.py
Comment (jwa7) [inline python/src/equistore/labels.py:284]: Did we remove the need to specify dtype here @Luthaf ?

**Review by jwa7 — CHANGES_REQUESTED** [review]
Looks mostly good! 

Quick comment: in your docstring examples (i.e. in solve.py) you include calls like:
```py
samples = Labels.arange(2, "sample")
```
But this raises an error as the function tries to interpret "sample" as an int:
```
ValueError: all numbers provided to `Labels.arange()` must be integers
```

These docstring examples will need updating to pass the name as a keyword arg.

Also, in this case, it might be helpful to receive an error message that points the user towards passing the name as a keyword argument instead of a positional one.

File: python/src/equistore/labels.py
Comment (jwa7) [inline python/src/equistore/labels.py:285]: Also, do we want to allow for creation of Labels of > 1 dimension?

Perhaps something like this, passing a list of names and using the length of this to reshape the np.arange array

```suggestion

        if isinstance(names, str):
            names = [names]
        if not np.all([isinstance(n, str) for n in names]):
            raise TypeError("all names provided to `Labels.arange()` must be strings")

        labels = Labels(
            names=names, values=np.arange(*args, dtype=np.int32).reshape(-1, len(names))
        )
```

File: python/src/equistore/labels.py
Comment (PicoCentauri) [reply python/src/equistore/labels.py:255]: And explicit parameters please. Who knows what you want as `*args`.

File: python/src/equistore/labels.py
Comment (PicoCentauri) [inline python/src/equistore/labels.py:281]: I think you don't need any checks here. Just pass everything to `np.arange`. They will handle it no?

File: python/tests/labels.py
Comment (PicoCentauri) [inline python/tests/labels.py:208]: You don't need to assign this.

**Comment by jwa7** [comment]
Re: my suggestions about multi-dimensonal - sorry, just seen the discussion on ML dev about keeping it to one dimension. Happy for it to be that way for now but think it'll be useful to have the multidimensional case eventually.

File: python/src/equistore/labels.py
Comment (PicoCentauri) [reply python/src/equistore/labels.py:284]: Maybe use `np.array(range(*args))[:, np.newaxis]`, then you are sure that everything is an integer.

File: python/src/equistore/labels.py
Comment (Luthaf) [reply python/src/equistore/labels.py:255]: I would force an explicit name instead of having a default:

```
    @staticmethod
    def arange(name, *args):
```

or with explicit arguments

```
    @staticmethod
    def arange(name, start, stop=None, step=None):
```

File: python/tests/labels.py
Comment (frostedoyster) [reply python/tests/labels.py]: As requested

File: python/tests/labels.py
Comment (frostedoyster) [reply python/tests/labels.py]: Ok, I've removed that and all the other discarded values in the other tests

File: python/src/equistore/labels.py
Comment (frostedoyster) [reply python/src/equistore/labels.py:289]: I feel like our own error messages can be more informative

File: python/src/equistore/labels.py
Comment (frostedoyster) [reply python/src/equistore/labels.py]: I've made the name mandatory and I've documented the three integer parameters

File: python/src/equistore/labels.py
Comment (frostedoyster) [reply python/src/equistore/labels.py]: I've tried to remove the np.int32 there and everywhere else in equistore

File: python/src/equistore/labels.py
Comment (PicoCentauri) [reply python/src/equistore/labels.py:290]: fine for me

**Review by PicoCentauri — APPROVED** [review]
Looks good and very nice that your removed the `np.int32` from the other files.

**Comment by frostedoyster** [comment]
@jwa7 Please don't merge... I messed up... There are so many more manual implementations that I missed simply because they weren't using np.arange

File: python/src/equistore/labels.py
Comment (DavideTisi) [inline python/src/equistore/labels.py:255]: one comment, nice function. For now the function just create a label with one name and only one arange.
Maybe it would be also helpful to have the possibility to create a label with more names, something like 

```python
arange(names=['a1', 'a2'],start=[0,3],stop=[2,7],step=[1,2])-> Labels(['a1', 'a2'],np.array([[0,3], [1,5]])```

the function should handle the errors if the the two arange are not compatible

File: python/src/equistore/labels.py
Comment (jwa7) [reply python/src/equistore/labels.py]: Decided to keep to one dimension

**Review by jwa7 — COMMENTED** [review]
Looking great, thanks @frostedoyster ! Also thanks for cleaning up the dtype stuff - much cleaner now. I removed a few remaining dtypes and merged master into here so we're ready to merge.

Before we merge, please could you address @DavideTisi 's comment above?

**Comment by DavideTisi** [comment]
@jwa7 but my comment was just to do the multi-dimensional thing. when I wrote the comment, I hadn't read the post on slack channel. 
My idea was basically just what Michele was suggesting there, but since it has been decided to do it only 1D for me it is ok.

For the rest it is super fine, I am not a super fan of having a function that wraps basically a single line, but I do not have much experience as a user, and the frustrations that come with it.

**Review by jwa7 — APPROVED** [review]

**Comment by jwa7** [comment]
OK thanks @DavideTisi - merged :)

File: python/src/equistore/labels.py
Comment (Luthaf) [inline python/src/equistore/labels.py:270]: This documentation indicate that users can call this function with `Labels.arange(name="s", start=3, stop=4)` which is not the case. We could also have the function take `*args, **kwargs` and check both, I would first check how numpy handles this.

Comment (frostedoyster) [reply python/src/equistore/labels.py:270]: I tried to check numpy's function arguments, but it seems to be written in C, so I gave up. I agree that using and checking kwargs is a good way to solve this. Probably also an example would help. I will work on this

**Pull request merged**
